### PR TITLE
Bug 1880004 - Add Sponsored Mobile Provider

### PR DESCRIPTION
**Pull request by tiftran**
Added amp mobile as a nimbus var provider, so that we can set it so fenix can query for amp suggestions that are specifically for mobile.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1880004

### CODE REVIEW

File: android-components/components/feature/fxsuggest/fxsuggest.fml.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ features:
         type: Map<SuggestionType, Boolean>
         default: {
           "amp": false,
+          "ampMobile": false,
```
Comment (MatthewTighe) [inline android-components/components/feature/fxsuggest/fxsuggest.fml.yaml:24]: Is it possible for these to both be true?

**Review by MatthewTighe — APPROVED** [review]
Had a question out of curiosity, but otherwise this LGTM! I'll let Lina handle the final approval

File: android-components/components/feature/fxsuggest/fxsuggest.fml.yaml (hunk 1 above)
Comment (tiftran) [reply android-components/components/feature/fxsuggest/fxsuggest.fml.yaml:24]: No, I don't believe both can be true at the same time

**Review by linabutler — APPROVED** [review]
Thanks @tiftran!

**Comment by tiftran** [comment]
@Mergifyio backport releases_v124

**Pull request merged**